packages/block-editor/src/components/inserter/utils.native.js

```diff
@@ -0,0 +1,28 @@
+const REUSABLE_BLOCKS_CATEGORY = 'reusable';
+const ALLOWED_EMBED_VARIATIONS = [ 'core/embed' ];
+
+export function blockAllowed( block, { onlyReusable, allowReusable } ) {
+	const { id, category } = block;
+	const isReusable = category === REUSABLE_BLOCKS_CATEGORY;
+
+	if ( onlyReusable ) {
+		return isReusable;
+	}
+
+	if ( isReusable ) {
+		return allowReusable;
+	}
+	// We don't want to show all possible embed variations
+	// as different blocks in the inserter. We'll only show a
+	// few popular ones.
+	return category !== 'embed' || ALLOWED_EMBED_VARIATIONS.includes( id );
+}
+
+export function filterInserterItems(
```

Comment:
It would be great if we could have a unit test to cover the logic of this function, wdyt?

Reply:
I'm going to merge this as-is and circle back to the unit test sometime next week. Thanks again for the review @fluiddot !